rust_src/src/marker.rs

```diff
@@ -127,6 +128,67 @@ pub fn marker_buffer(marker: LispMarkerRef) -> Option<LispBufferRef> {
     marker.buffer()
 }
 
+/// Return a newly allocated marker which points into BUF
+/// at character position CHARPOS and byte position BYTEPOS.
+#[no_mangle]
+pub extern "C" fn build_marker(
+    buf: *mut Lisp_Buffer,
+    charpos: ptrdiff_t,
+    bytepos: ptrdiff_t,
+) -> LispObject {
+    debug_assert!(unsafe { (*buf).name_.is_not_nil() });
+    debug_assert!(charpos <= bytepos);
+
+    unsafe {
+        let obj = allocate_misc(Lisp_Misc_Type::Lisp_Misc_Marker);
+        let mut m = obj.as_marker_or_error();
+
+        m.set_buffer(buf);
+        m.set_charpos(charpos);
+        m.set_bytepos(bytepos);
+        m.set_insertion_type(false);
+        m.set_need_adjustment(false);
+
+        let mut buffer_ref = LispBufferRef::from_ptr(buf as *mut c_void)
+            .unwrap_or_else(|| panic!("Invalid buffer reference."));
+
+        m.set_next((*buffer_ref.text).markers);
+        (*buffer_ref.text).markers = m.as_mut();
+
+        obj
+    }
+}
+
+/// Return value of point, as a marker object.
+#[lisp_fn]
+pub fn point_marker() -> LispObject {
+    unsafe {
+        let cur_buf = (*current_thread).m_current_buffer;
+        build_marker(cur_buf, (*cur_buf).pt, (*cur_buf).pt_byte)
```

Comment:
I missed this the first time, but you can simplify this by using `ThreadState::current_buffer()`.  Looks pretty good, otherwise.